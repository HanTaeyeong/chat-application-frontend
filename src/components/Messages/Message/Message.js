import React, { useState } from 'react';
import './Message.scss'

function Message({ name, message }) {
    const { user, text } = message || {};
    const isSentByCurrentUser = (user === name);

    return (
        <div className={`message-container flex-${isSentByCurrentUser ? 'end' : 'start'}`}>
            <div>
                {!isSentByCurrentUser && <p className={`sentText`}>{user}</p>}
            </div>
            <div className={`message-box background-${isSentByCurrentUser ? 'blue' : 'light'}`}>
                <p className={`message-text color-${isSentByCurrentUser ? 'white' : 'dark'}`}>{text}</p>
            </div>
        </div>);
}



export default Message;
