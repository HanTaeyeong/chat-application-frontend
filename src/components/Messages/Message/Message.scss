.message-box {
  background: #f3f3f3;
  border-radius: 20px;
  padding: 5px 20px;
  color: white;
  display: inline-block;
  max-width: 80%;
}

.message-text {
  width: 100%;
  letter-spacing: 0;
  float: left;
  font-size: 1.1em;
  word-wrap: break-word;
}

.message-text img {
  vertical-align: middle;
}

.message-container {
  display: flex;
  flex-direction: column;
  justify-content:center;

  padding: 0 5%;
  margin-top: 3px;

  &.flex-start{
    align-items: flex-start;
  }
  &.flex-end{
    align-items: flex-end;
  }
  
}

.sent-text {
  display: flex;
  align-items: center;
  font-family: Helvetica;
  color: #828282;
  letter-spacing: 0.3px;

}

.color-white {
  color: white;
}

.color-dark {
  color: #353535;
}

.background-blue {
  background: #2979ff;
}

.background-light {
  background: #f3f3f3;
}
