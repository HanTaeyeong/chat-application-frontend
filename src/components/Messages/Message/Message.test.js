import React from 'react'
import Message from './Message'

import {create} from 'react-test-renderer'

describe('Message Component Test', () => {
    test('render test', () => {   
        let app=create(<Message />) 
        expect(app.toJSON()).toMatchSnapshot()
    })
})