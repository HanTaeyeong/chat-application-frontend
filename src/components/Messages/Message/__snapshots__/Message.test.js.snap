// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component Test render test 1`] = `
<div
  className="message-container flex-end"
>
  <div />
  <div
    className="message-box background-blue"
  >
    <p
      className="message-text color-white"
    />
  </div>
</div>
`;
